x-pack/plugin/watcher/qa/common/src/main/java/org/elasticsearch/xpack/watcher/WatcherRestTestCase.java

```diff
@@ -70,18 +75,23 @@ public final void stopWatcher() throws Exception {
                     throw new AssertionError("unknown state[" + state + "]");
             }
         }, 60, TimeUnit.SECONDS);
+        deleteAllWatcherData();
+    }
+
+    public static void deleteAllWatcherData() throws IOException {
+        var queryWatchesRequest = new Request("GET", "/_watcher/_query/watches");
+        var response = ObjectPath.createFromResponse(ESRestTestCase.adminClient().performRequest(queryWatchesRequest));
 
-        Request deleteWatchesIndexRequest = new Request("DELETE", ".watches");
-        deleteWatchesIndexRequest.addParameter("ignore_unavailable", "true");
-        deleteWatchesIndexRequest.setOptions(
-            expectWarnings(
-                "this request accesses system indices: [.watches], but in a future major "
-                    + "version, direct access to system indices will be prevented by default"
-            )
-        );
-        ESRestTestCase.adminClient().performRequest(deleteWatchesIndexRequest);
+        int totalCount = response.evaluate("count");
+        List<Map<?, ?>> watches = response.evaluate("watches");
+        assertThat("Less watches requested than exist in total", watches.size(), equalTo(totalCount));
```

Comment:
nit: I assume this would be a programmers error, if so can this just be a Java `assert` instead of an `assertThat` to help differentiate between something that is being tested and a self documenting sanity check. Also, why Less ? should be it be not equal ?

Reply:
Right, I will adjust the worden and turn this into an `assert`.